src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractCohortEngine.java

```diff
@@ -656,6 +613,49 @@ private VariantContext createVariantContextFromSampleRecord(final ExtractCohortR
                             .map(Integer::parseInt)
                             .map(alleleIndex -> alleles.get(alleleIndex))
                             .collect(Collectors.toList());
+
+            if (performGenotypeVQSLODFiltering) {
+                final List<Allele> nonRefAlleles =
+                        genotypeAlleles.stream()
+                                .filter(a -> a.isNonReference())
+                                .distinct()
+                                .collect(Collectors.toList());
+
+                final LinkedHashMap<Allele, Double> remappedVqsLodMap = remapAllelesInMap(ref, nonRefAlleles, contig, (int) startPosition, vqsLodMap, Double.NaN);
+                final LinkedHashMap<Allele, String> remappedYngMap = remapAllelesInMap(ref, nonRefAlleles, contig, (int) startPosition, yngMap, VCFConstants.EMPTY_INFO_FIELD);
+
+                // see https://github.com/broadinstitute/dsp-spec-ops/issues/291 for rationale
+                // take "worst" outcome for yng/vqslod, evaluate each allele separately
+                // if any allele is "N"ay, the genotype is filtered
+                // if any allele is "Y"ay and the rest are "G"rey, the genotype is passed
+                // if all alleles are "G"ray, the VQSLod is evaluated
+                boolean anyNays = nonRefAlleles.stream().map(a -> remappedYngMap.get(a)).anyMatch(v -> "N".equals(v));
+                boolean anyYays = nonRefAlleles.stream().map(a -> remappedYngMap.get(a)).anyMatch(v -> "Y".equals(v));
+
+                // if there are any "N"s, the genotype is filtered
+                if (anyNays) {
+                    genotypeBuilder.filter(GATKVCFConstants.NAY_FROM_YNG);
+                } else if (anyYays) {
+                    // the genotype is passed, nothing to do here as non-filtered is the default
+                } else {
+                    // get the minimum (worst) vqslod for all SNP non-Yay sites, and apply the filter
+                    Optional<Double> snpMin =
```

Comment:
we are actually getting the max. if that is what we want, update the comment and variable name for this and indels

Reply:
good catch -- updating